### ISSUE TITLE
Property __get__ usage causes E1101 (no-member) error

### ISSUE THREAD

**Issue by cossor** [PyCQA/pylint#4817]
### Bug description

```python
"""
NB: i am using pylint 2.6.2, because pip won't install a newer one!

too often, i've written a comment indicating that a method that would normally
be a property is a method because it is used as a sort key. after discovering
that there's an easy way to __get__ properties via function calls, i started
to use this technique, mostly to see whether doing so helps.

(on the one hand, most people aren't used to seeing this usage. on the other
hand, its meaning should be clear? in general, it's good to make a change like
this in more than a few places before evaluating it.)

unfortunately, pylint doesn't know about this __get__ method, so my experiment
includes adding comments to disable pylint complaints.
"""


# pylint: disable=missing-class-docstring,missing-function-docstring

class Spam:
    def __init__(self, key: int):
        self._key = key

    def __repr__(self):
        return "Spam({})".format(self.key)

    @property
    def key(self) -> int:
        return self._key


if __name__ == "__main__":
    spam = [Spam(0), Spam(1)]
    spam.sort(key=repr, reverse=True)
    print(spam)  # [Spam(1), Spam(0)]

    assert "__get__" in dir(Spam.key)
    spam.sort(key=Spam.key.__get__)
    print(spam)  # [Spam(0), Spam(1)]
```

### Command used

```shell
pylint spam.py
```

### Pylint output

```shell
************* Module spam
spam.py:38:18: E1101: Function 'key' has no '__get__' member (no-member)
```

### Expected behavior

pylint should not complain about Spam.key.\_\_get\_\_.

### Pylint version

```shell
pylint 2.6.2
astroid 2.4.2
Python 3.5.2 (default, Jan 26 2021, 13:30:48) 
[GCC 5.4.0 20160609]
```

### OS / Environment

Ubuntu 16.04 (Xenial)

**Comment by Pierre-Sassoulas** [comment]
Sorry we're not maintaining 2.6 anymore and this is fixed in the latest version released.

**Comment by cossor** [comment]
no apology needed. thanks for the good news!
